Source/DafnyLanguageServer/Workspace/DocumentDatabase.cs

```diff
@@ -99,65 +109,107 @@ ISymbolTableRelocator symbolTableRelocator
       }
       databaseEntry.CancelPendingUpdates();
       var cancellationSource = new CancellationTokenSource();
+      var previousDocumentTask = FirstSuccessfulTask(databaseEntry.VerifiedDocument, databaseEntry.ResolvedDocument);
+      var resolvedDocumentTask = ApplyChangesAsync(previousDocumentTask, documentChange, cancellationSource.Token);
+      var verifiedDocument = VerifyAsync(resolvedDocumentTask, VerifyOnChange, cancellationSource.Token);
       var updatedEntry = new DocumentEntry(
         documentChange.TextDocument.Version,
-        Task.Run(async () => await ApplyChangesAsync(await databaseEntry.Document, documentChange, cancellationSource.Token)),
+        resolvedDocumentTask,
+        verifiedDocument,
         cancellationSource
       );
       documents[documentUri] = updatedEntry;
-      return await updatedEntry.Document;
+      return resolvedDocumentTask.ToObservable().Concat(verifiedDocument.ToObservable());
+    }
+
+
+#pragma warning disable VSTHRD200
+    private static Task<T> FirstSuccessfulTask<T>(params Task<T>[] tasks) {
+#pragma warning restore VSTHRD200
+      var taskList = tasks.ToList();
+      var tcs = new TaskCompletionSource<T>();
+      int remainingTasks = taskList.Count;
+      foreach (var task in taskList) {
+#pragma warning disable VSTHRD110
```

Comment:
I believe you can remove this warning by assigning to a forgetting pattern instead of pragma directives, like that:

    _ = task.ContinueWith( ...

See https://stackoverflow.com/questions/14903887/warning-this-call-is-not-awaited-execution-of-the-current-method-continues